File: common/src/main/java/org/tdf/common/Context.java
package org.tdf.common;

import java.util.Collection;

// context for communicating with peer server and listener
public interface Context {
    // exit listeners chain
    void exit();

    // disconnect to the peer
    void disconnect();

    // block the peer for a while
    void block();

    // keep the connection alive
    void keep();

    // response to the remote peer
    void response(byte[] message);

    // batch response
    void response(Collection<byte[]> messages);

    // relay the received message
    void relay();

    // get the message received from channel
    byte[] getMessage();

    // get remote peer
    Peer getRemote();
}

File: facade/src/main/java/org/tdf/sunflower/net/Context.kt
package org.tdf.sunflower.net

// context for communicating with peer server and listener
interface Context {
    // exit listeners chain
    fun exit()

    // disconnect to the peer
    fun disconnect()

    // block the peer for a while
    fun block()

    // keep the connection alive
    fun keep()

    // response to the remote peer
    fun response(message: ByteArray)

    // batch response
    fun response(messages: Collection<ByteArray>)

    // relay the received message
    fun relay()

    // get the message received from channel
    val message: ByteArray

    // get remote peer
    val remote: Peer

    val udp: Boolean
}